From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Thuong Vo <66083331+SekiroKenjii@users.noreply.github.com>
Date: Sat, 24 Oct 2026 07:51:56 +0000
Subject: [PATCH] build: compile the Apex resource into nautilus

The Apex Files widgets ship inside nautilus. Their sources come from
Source1, a git archive of desktop/nautilus/ in apex-gnome, which %prep
unpacks at the top of this tree as src/apex/. src/meson.build reads
src/apex/meson.build right after upstream's sources, which adds the Apex
sources and resource, /org/gnome/nautilus/apex, to libnautilus.
main() registers that resource next to upstream's, as nothing else
links it in from the static library.
---
 src/meson.build     | 1 +
 src/nautilus-main.c | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/src/meson.build b/src/meson.build
index c15e6ce..718d5d3 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -288,6 +288,7 @@ libnautilus_sources = [
   'nautilus-x-content-bar.c',
   'nautilus-x-content-bar.h',
 ]
+subdir('apex')
 
 nautilus_deps = [
   cloudproviders,
diff --git a/src/nautilus-main.c b/src/nautilus-main.c
index 3ca374e..98fd7b8 100644
--- a/src/nautilus-main.c
+++ b/src/nautilus-main.c
@@ -29,6 +29,7 @@
 
 #include "nautilus-application.h"
 #include "nautilus-resources.h"
+#include "apex/apex.h"
 
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
@@ -65,6 +66,7 @@ main (int   argc,
     }
 
     nautilus_register_resource ();
+    apex_files_init ();
     /* Run the nautilus application. */
     application = nautilus_application_new ();
 
-- 
2.43.0

